src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColExtendedDataService.java

```diff
@@ -0,0 +1,160 @@
+package uk.ac.ebi.eva.evaseqcol.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.ac.ebi.eva.evaseqcol.entities.AssemblyEntity;
+import uk.ac.ebi.eva.evaseqcol.entities.AssemblySequenceEntity;
+import uk.ac.ebi.eva.evaseqcol.entities.ChromosomeEntity;
+import uk.ac.ebi.eva.evaseqcol.entities.SeqColEntity;
+import uk.ac.ebi.eva.evaseqcol.entities.SeqColExtendedDataEntity;
+import uk.ac.ebi.eva.evaseqcol.entities.SeqColSequenceEntity;
+import uk.ac.ebi.eva.evaseqcol.entities.SequenceEntity;
+import uk.ac.ebi.eva.evaseqcol.repo.SeqColExtendedDataRepository;
+import uk.ac.ebi.eva.evaseqcol.utils.JSONExtData;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class SeqColExtendedDataService {
+
+    @Autowired
+    private SeqColExtendedDataRepository repository;
+
+    public Optional<SeqColExtendedDataEntity> addSeqColExtendedData(SeqColExtendedDataEntity seqColExtendedData){
+        try {
+            SeqColExtendedDataEntity seqCol = repository.save(seqColExtendedData);
+            return Optional.of(seqCol);
+        } catch (Exception e){
+            // TODO : THROW A SELF MADE EXCEPTION
+            e.printStackTrace();
+            //System.out.println("SeqcolL2 with digest " + seqColExtendedData.getDigest() + " already exists in the db !!");
+
+        }
+        return Optional.empty();
+    }
+
+    @Transactional
+    public List<SeqColExtendedDataEntity> addAll(List<SeqColExtendedDataEntity> seqColExtendedDataList) {
+        return repository.saveAll(seqColExtendedDataList);
+    }
+
+    public List<SeqColExtendedDataEntity> getAll() {
+        return repository.findAll();
+    }
+
+    /**
+     * Return the 3 extracted seqcol objects (names, lengths and sequences) of the given naming convention*/
+     public List<SeqColExtendedDataEntity> constructLevelTwoSeqCols(AssemblyEntity assemblyEntity, AssemblySequenceEntity sequenceEntity,
+                                                                    SeqColEntity.NamingConvention convention, String accession){
+        // Sorting the chromosomes' list (assemblyEntity) and the sequences' list (sequencesEntity) in the same order
+        sortReportAndSequencesByRefseq(assemblyEntity, sequenceEntity, accession);
```

Comment:
Actually I wasn't aware that the original ordering of the sequences that comes in the assembly report and in the FASTA file should be kept identical when creating seqCol objects.

And the reason why I attempted to sort the sequences before processing them, is to prevent any problem when the sequences in the assembly report and the ones in the FASTA file did not come in the same order, which will cause collision.

Currently, all the files that I've processed so far contained sequences ordered in the same order, however I did the sort because I couldn't make sure that all the files that will be processed in the future will respect that